app/lib/screens/notifications_screen.dart

```diff
@@ -12,37 +12,63 @@ class NotificationsScreen extends StatefulWidget {
 }
 
 class _NotificationsScreenState extends State<NotificationsScreen> {
+  late NotificationService _notificationService;
+
+  @override
+  void initState() {
+    super.initState();
+    _notificationService = GetIt.I<NotificationService>();
+    _notificationService.addListener(_onNotificationsUpdated);
+  }
+
+  void _onNotificationsUpdated() {
+    setState(() {});
```

Comment:
Instead of using `setState `and potentially rebuilding the whole page you could also use a `ListenableBuilder` and pass `NotificationService` to it as it is a `ChangeNotifier`. Similiar to a `FutureBuilder` it rebuilds according to the function on changes.  
There is also a `ValueListenableBuilder` which is basically the same but also provides a value.  
I found those mechanics to be great way to update the UI.  
https://api.flutter.dev/flutter/widgets/ListenableBuilder-class.html
https://api.flutter.dev/flutter/foundation/ValueListenable-class.html
However i don't think a change is needed here, just thought it might be interesting ^^